Makefile

```diff
@@ -60,7 +60,11 @@ IMAGE_ARCH_SUFFIX = $(addprefix -,$(filter $(ARCH),arm64))
 # GOLANG_IMAGE is the building golang container image used.
 GOLANG_IMAGE = public.ecr.aws/docker/library/golang:1.16-stretch
 # For the requested build, these are the set of Go specific build environment variables.
-export GOARCH ?= $(ARCH)
+ifdef TARGETARCH
```

Comment:
This works, but adds complexity and extra codepaths.  Can we just pass `make ARCH=$TARGETARCH` from the Dockerfile?  (or `make GOARCH=$TARGETARCH` ?  There are already too many $arch variables in this Makefile :confused: )

Reply:
Should we remove line 54?
https://github.com/aws/amazon-vpc-cni-k8s/blob/050d9be772edc1c8fa6dc2548e0fc98aa128c952/Makefile#L54

And just replace all this with `export GOARCH = $(ARCH)` - with the idea that $(ARCH) is usually unset.

I think if we _don't_ set GOARCH then it defaults to the current architecture, and all the UNAME_ARCH/ARCH stuff above is just a complicated way of doing the same thing.

... Alternatively, rip out everything to do with $(ARCH), $(GOARCH), and $(GOOS) in this file.  Then expect the caller (ie: Dockerfile) to set GOOS/GOARCH before invoking this Makefile, if cross-compiling.

Basically, we should set the target GOOS/GOARCH in one layer - either the Makefile or the Dockerfile.  And not some fight between both.

Reply:
Also we can remove all the IMAGE_ARCH_SUFFIX stuff now that we have true multi-arch images:
https://github.com/aws/amazon-vpc-cni-k8s/blob/050d9be772edc1c8fa6dc2548e0fc98aa128c952/Makefile#L59

Reply:
@anguslees Thanks for the feedback, I will work on removing the extra variables for `IMAGE_ARCH_SUFFIX`.

Regarding the switch statement in Makefile, it's currently needed as docker platform arguments such as `TARGETARCH` are set only when buildkit is used (Reference: From https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope, `This feature is only available when using the BuildKit backend`). 

Further, docker buildkit is not yet the default in linux (https://github.com/moby/moby/issues/40379). As such, to make the build process work in scenarios where buildkit isn't used, setting the arch variable in makefile is needed.

Reply:
`I think if we don't set GOARCH then it defaults to the current architecture, and all the UNAME_ARCH/ARCH stuff above is just a complicated way of doing the same thing.` 
That's correct and the build itself would succeed even if GOARCH isn't set. However, it would not be able to identify the correct version of networking plugin to download here - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/Makefile#L211

In case of buildkit driven builds, this would need to be derived from the TARGETARCH platform argument and in case of non buildkit builds, it would have to be derived from `uname` output.

Reply:
With the modified approach of setting buildkit at the outset in makefile with `export DOCKER_BUILDKIT=1` https://github.com/aws/amazon-vpc-cni-k8s/pull/1824/files, I have removed the extra executions for `UNAME_ARCH` and `ARCH`.

Now, value for `GOARCH` variable is directly derived from the docker platform arg value of `TARGETARCH`